src/main/java/duke/commands/AddCommand.java

```diff
@@ -0,0 +1,158 @@
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private DateTime dateTime = new DateTime();
+    private String[] allDetails;
+
+    /**
+     * Loads the entire input command into the object for further processing of its details, such as
+     * type of task added and other details (e.g. description and time).
+     * @param allDetails a String array containing all the details to create the appropriate task
+     */
+    public AddCommand(String[] allDetails) {
+        this.allDetails = allDetails;
+    }
+
+    /**
+     * Checks for the type of task that has been added and passes along processed String of information
+     * containing details about the task for the task to be created.
+     * @param tasks contains the data structure of Tasks stored in Duke
+     * @param ui contains methods dealing with interaction with the user
+     * @param storage contains methods to load and save information in the file
+     * @exception DukeException is thrown when there is an error with the input
+     * @exception IOException is thrown when there is an error saving the data in the hard disk
+     */
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException, IOException {
+        this.tasks = tasks;
+        this.ui = ui;
+        this.storage = storage;
+        String processed = "";
+        for (int i = 1; i < allDetails.length; i++) {
+            processed += allDetails[i] + " ";
+        }
+        String commandWord = allDetails[0];
+        if (allDetails.length == 1) {
+            throw new DukeException("Oops! The description of your Task cannot be empty.");
+        } else if (commandWord.equals("todo")) {
+            return toDoTask(processed);
+        } else if (commandWord.equals("deadline")) {
+            return deadlineTask(processed);
+        } else {
+            return eventTask(processed);
+        }
+    }
+
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Creates a TD object and adds it into memory.
+     * @param a a String of information containing details for the To Do task
+     * @exception IOException is thrown when there is an error saving the data in the hard disk
+     */
+    private String toDoTask(String a) throws IOException {
+        ToDo newTodo = new ToDo(a);
+        this.tasks.addTask(newTodo);
+        this.storage.save(tasks);
+        return printAddedTask(newTodo);
+    }
+
+    /**
+     * Creates a deadlineTask and adds it into memory.
+     * If command does not include a deadline, then user is prompted to enter the command again.
+     * @param b a String of information containing details for the Deadline task
+     * @exception DukeException is thrown when there is an error with the input
+     * @exception IOException is thrown when there is an error saving the data in the hard disk
+     */
+    private String deadlineTask(String b) throws DukeException, IOException {
```

Comment:
I feel like this should belong to some Parser.class? Or at least be abstracted to check whether it's a valid input first.